backend/src/main/java/ai/verta/modeldb/configuration/AppConfigBeans.java

```diff
@@ -0,0 +1,412 @@
+    }
+
+    // Initialize data access
+    return DAOSet.fromServices(services, config.getJdbi(), handleExecutor, config);
+  }
+
+  @Bean
+  public ArtifactStoreService artifactStoreService(MDBConfig config) throws IOException {
+
+    final var artifactStoreConfig = config.artifactStoreConfig;
+    if (artifactStoreConfig.isEnabled()) {
+      //
+      // TODO: This is backwards, these values can be extracted from the environment or injected
+      // using profiles instead
+      // ------------- Start Initialize Cloud storage base on configuration ------------------
+      ArtifactStoreService artifactStoreService;
+
+      if (artifactStoreConfig.getArtifactEndpoint() != null) {
+        System.getProperties()
+            .put(
+                "artifactEndpoint.storeArtifact",
+                artifactStoreConfig.getArtifactEndpoint().getStoreArtifact());
+        System.getProperties()
+            .put(
+                "artifactEndpoint.getArtifact",
+                artifactStoreConfig.getArtifactEndpoint().getGetArtifact());
+      }
+
+      if (artifactStoreConfig.getArtifactStoreType().equals("NFS")
+          && artifactStoreConfig.getNFS() != null
+          && artifactStoreConfig.getNFS().getArtifactEndpoint() != null) {
+        System.getProperties()
+            .put(
+                "artifactEndpoint.storeArtifact",
+                artifactStoreConfig.getNFS().getArtifactEndpoint().getStoreArtifact());
+        System.getProperties()
+            .put(
+                "artifactEndpoint.getArtifact",
+                artifactStoreConfig.getNFS().getArtifactEndpoint().getGetArtifact());
+      }
+
+      switch (artifactStoreConfig.getArtifactStoreType()) {
+        case "S3":
+          if (!artifactStoreConfig.S3.getS3presignedURLEnabled()) {
+            appContext.registeredBean("s3Controller", S3Controller.class);
+          }
+          artifactStoreService = new S3Service(artifactStoreConfig.S3.getCloudBucketName());
+          break;
+        case "NFS":
+          appContext.registeredBean("nfsController", NFSController.class);
+          String rootDir = artifactStoreConfig.getNFS().getNfsRootPath();
+          LOGGER.trace("NFS server root path {}", rootDir);
+          final var props = new FileStorageProperties();
+          props.setUploadDir(rootDir);
+          artifactStoreService = new NFSService(props);
+          break;
+        default:
+          throw new ModelDBException("Configure valid artifact store name in config.yaml file.");
+      }
+      // ------------- Finish Initialize Cloud storage base on configuration ------------------
+
+      LOGGER.info(
+          "ArtifactStore service initialized and resolved storage dependency before server start");
+      return artifactStoreService;
+    } else {
+      LOGGER.info("Artifact store service is disabled.");
+      return null;
+    }
+  }
+
+  public static boolean migrate(MDBConfig mdbConfig, JdbiUtil jdbiUtil, Executor handleExecutor)
```

Comment:
this shouldn't be in a file that contains bean definitions. It definitely deserves its own class.

Reply:
Resolved